src/main/java/org/springframework/data/redis/connection/lettuce/LettuceConnectionFactory.java

```diff
@@ -279,6 +296,32 @@ public LettuceConnectionFactory(RedisClusterConfiguration clusterConfiguration,
 		this.configuration = clusterConfiguration;
 	}
 
+	/**
+	 * Converts a {@link RedisURI} into its corresponding {@link RedisConfiguration} according to the following:
+	 * <ul>
+	 * <li>If {@code redisURI} has sentinel info a {@link RedisSentinelConfiguration} is returned</li>
+	 * <li>If {@code redisURI} has socket info a {@link RedisSocketConfiguration} is returned</li>
+	 * <li>Otherwise a {@link RedisStandaloneConfiguration} is returned</li>
+	 * </ul>
+	 *
+	 * @param redisURI the connection info in the format of a RedisURI
+	 * @return an appropriate {@link RedisConfiguration} instance representing the Redis URI.
+	 */
+	public static RedisConfiguration redisConfigurationFromRedisUri(RedisURI redisURI) {
+
+		Assert.notNull(redisURI, "RedisURI must not be null");
+
+		if (!ObjectUtils.isEmpty(redisURI.getSentinels())) {
+			return LettuceConverters.redisUriToSentinelConfiguration(redisURI);
```

Comment:
I went ahead and followed suit w/ LettuceConverters housing the conversions as it is the reverse direction of the `sentinel config to redis uri` that was already in there. I am happy moving elsewhere if need be.